src/hotspot/share/runtime/flags/jvmFlag.hpp

```diff
@@ -105,12 +106,36 @@ struct JVMFlag {
     NOTPRODUCT_FLAG_BUT_PRODUCT_BUILD
   };
 
-  const char* _type;
-  const char* _name;
+#define JVM_FLAG_NON_STRING_TYPES_DO(f) \
+    f(bool) \
+    f(int) \
+    f(uint) \
+    f(intx) \
+    f(uintx) \
+    f(uint64_t) \
+    f(size_t) \
+    f(double)
+
+#define JVM_FLAG_TYPE_DECLARE(t) \
+  TYPE_ ## t,
+
+  enum FlagType : int {
+    JVM_FLAG_NON_STRING_TYPES_DO(JVM_FLAG_TYPE_DECLARE)
+    // The two string types are a bit irregular: is_ccstr() returns true for both types.
+    TYPE_ccstr,
+    TYPE_ccstrlist,
+    NUM_FLAG_TYPES
+  };
```

Comment:
Are those macros really needed here and better than simply saying:

  enum FlagType : int {
    TYPE_bool,
    TYPE_int,
    TYPE_uint,
    TYPE_intx,
    TYPE_uintx,
    TYPE_uint64_t,
    TYPE_size_t,
    TYPE_double,
    TYPE_ccstr,
    TYPE_ccstrlist,
    NUM_FLAG_TYPES
  };

Reply:
`JVM_FLAG_NON_STRING_TYPES_DO` is used in several places, such as declaring functions like `is_int()`, `get_int()`, `set_int()`, etc. I think it's less tedious/error prone than writing the same code 9 times.